src/Microsoft.OData.Core/JsonLight/ODataJsonLightDeserializer.cs

```diff
@@ -258,7 +258,8 @@ internal static bool TryParsePropertyAnnotation(string propertyAnnotationName, o
             ODataPayloadKind payloadKind,
             PropertyAndAnnotationCollector propertyAndAnnotationCollector,
             bool isReadingNestedPayload,
-            bool allowEmptyPayload)
+            bool allowEmptyPayload,
+            string navigationSource = null)

```

Comment:
Why not make this `IEdmNavigationSource navigationSource` instead of `string navigationSource`.